packages/headless/src/controllers/commerce/context/cart/headless-cart.ts

```diff
@@ -37,6 +37,11 @@ export interface CartItem {
    */
   productId: string;
 
+  /**
+   * The unique identifier of the product variant.
+   */
+  sku: string;
```

Comment:
Should we make this optional? Sometimes product's don't have variants. We could fallback to the `productId` when the `sku` isn't provided.

Reply:
Interesting point. The data guru of Commerce should validate that.
My guess/take is that "A product MUST have a variant"

"Any color the customer wants, as long as it's black."

Reply:
I've updated the description to indicates that IF the cart item is not a variant, then set the same value as the product ID. 

Technically, not all products has variants (you could have a catalog that has only products), so it's true that the `sku` might be irrelevant in these cases. 

I'm open to set as optional, but my main concern is that people could forget to set this attribute because of it (resulting of bad attribution). If you set this attribute as required, but specify that it could be the same value as `productId`, I _think_ it's less prone to implementation error.

Reply:
> My guess/take is that "A product MUST have a variant"

It's possible to configure a catalog without variants, see https://docs.coveo.com/en/3139/coveo-for-commerce/commerce-catalog#add-a-catalog